appengine/src/main/java/com/google/googleinterns/gscribe/services/impl/SpreadsheetServiceImpl.java

```diff
@@ -0,0 +1,271 @@
+     * @param examMetadata ( exam metadata for exam to get spreadsheet id )
+     * @throws GeneralSecurityException,IOException ( thrown by NetHttpTransport, GoogleClientSecrets, GoogleTokenResponse or by invalid credentials file  )
+     */
+    private void addResponseToSheet(ExamInstance examInstance, User user, ExamMetadata examMetadata) throws GeneralSecurityException, IOException {
+        final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+        final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+        Credential credential = new Credential(BearerToken.authorizationHeaderAccessMethod()).setAccessToken(user.getAccessToken());
+        Sheets service = new Sheets.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential).setApplicationName("Gscribe").build();
+
+        List<Object> response = new ArrayList<>();
+        response.add(examInstance.getStartTime().toString());
+        response.add(Integer.toString(examInstance.getStudentRollNum()));
+        for (int i = 0; i < examInstance.getAnswers().getAnswersList().size(); i++) {
+            response.add(examInstance.getAnswers().getAnswersList().get(i).getAnswer());
+        }
+        response.add("-");
+        List<List<Object>> toWrite = new ArrayList<>();
+        toWrite.add(response);
+        ValueRange body = new ValueRange().setValues(toWrite);
+        String spreadSheetID = examMetadata.getSpreadsheetID();
+        String sheetName = "Responses_" + examInstance.getExamID();
+        service.spreadsheets().values().append(spreadSheetID, sheetName, body).setValueInputOption("USER_ENTERED").execute();
+    }
+
+    /**
+     * Called to store the response from examinee to the responses spreadsheet
+     * Forms the row to be added to spreadsheet and adds it to spreadsheet via addResponseToSheet function
+     * If first time gets 401 error then refresh the token as access token might have expired
+     *
+     * @param examInstance ( exam instance object for the exam taken )
+     * @param user         ( paper setter user object to access spreadsheet )
+     * @param examMetadata ( exam metadata for exam to get spreadsheet id )
+     * @throws GeneralSecurityException,IOException ( thrown by NetHttpTransport, GoogleClientSecrets, GoogleTokenResponse or by invalid credentials file  )
+     * @throws InvalidDatabaseDataException         ( If the user tokens stored in database are malformed )
+     * @throws InvalidRequestException              ( If unable to access spreadsheet to submit response )
+     */
+    @Override
+    public void addResponseRequest(ExamInstance examInstance, User user, ExamMetadata examMetadata) throws GeneralSecurityException, InvalidDatabaseDataException, InvalidRequestException, IOException {
+        try {
+            addResponseToSheet(examInstance, user, examMetadata);
+        } catch (GoogleJsonResponseException e) {
+            if (e.getStatusCode() == 401) {
```

Comment:
Separate this out into another function.

Reply:
Created new function for this.